packages/eas-cli/src/prompts.ts

```diff
@@ -9,87 +9,128 @@ export interface ExpoChoice<T> extends Choice {
 
 type NamelessQuestion = Omit<Question<'value'>, 'name' | 'type'>;
 
-export async function promptAsync<T extends string = string>(
-  questions: Question<T> | Question<T>[],
-  options: Options = {}
-): Promise<Answers<T>> {
-  if (!process.stdin.isTTY && !global.test) {
-    throw new Error('Input is required, but stdin is not readable.');
-  }
-  return await prompts<T>(questions, {
-    onCancel() {
-      process.exit(constants.signals.SIGINT + 128); // Exit code 130 used when process is interrupted with ctrl+c.
-    },
-    ...options,
-  });
+export interface Prompts {
+  promptAsync<T extends string = string>(
+    questions: Question<T> | Question<T>[],
+    options?: Options
+  ): Promise<Answers<T>>;
+  confirmAsync(question: NamelessQuestion, options?: Options): Promise<boolean>;
+  selectAsync<T>(message: string, choices: ExpoChoice<T>[], options?: Options): Promise<T>;
+  toggleConfirmAsync(questions: NamelessQuestion, options?: Options): Promise<boolean>;
+  pressAnyKeyToContinueAsync(): Promise<void>;
 }
 
-export async function confirmAsync(
-  question: NamelessQuestion,
-  options?: Options
-): Promise<boolean> {
-  const { value } = await promptAsync(
-    {
-      initial: true,
-      ...question,
-      name: 'value',
-      type: 'confirm',
-    },
-    options
-  );
-  return value;
-}
-export async function selectAsync<T>(
-  message: string,
-  choices: ExpoChoice<T>[],
-  options?: Options
-): Promise<T> {
-  const { value } = await promptAsync(
-    {
-      message,
-      choices,
-      name: 'value',
-      type: 'select',
-    },
-    options
-  );
-  return value ?? null;
-}
+export class NonInteractivePrompts implements Prompts {
+  private static errorMessage =
+    'Command can not be run as specified in non-interactive mode. Run the command with --help to see arguments and flags required to run in non-interactive mode.';
+
+  async promptAsync<T extends string = string>(
```

Comment:
If we have a bug where we should handle non-interactive mode but for some reason we did not, the user will see just that prompt. Without any context on what question was the problem, it will be hard to debug. I think we should also print a question in non-interactive mode.

Reply:
The idea is that we hopefully catch these cases ahead of releasing it and surround them with conditionals and better thrown errors. (see the example converted commands)